src/util.cpp

```diff
@@ -167,23 +167,47 @@ instance_of_cinit;
  */
 
 static boost::once_flag debugPrintInitFlag = BOOST_ONCE_INIT;
+
 /**
- * We use boost::call_once() to make sure these are initialized
- * in a thread-safe manner the first time called:
+ * We use boost::call_once() to make sure mutexDebugLog and
+ * vMsgsBeforeOpenLog are initialized in a thread-safe manner.
+ *
+ * FIXME: fileout, mutexDebugLog and vMsgsBeforeOpenLog are leaked
+ * on exit.  This is ugly, but will be cleaned up by the OS/libc.
+ * When the shutdown sequence is fully audited and tested, explicit
+ * destruction of these objects can be implemented.
  */
 static FILE* fileout = NULL;
 static boost::mutex* mutexDebugLog = NULL;
+static list<string> *vMsgsBeforeOpenLog;
+
+int LogWriteStr(const std::string &str)
```

Comment:
Also I'd suggest to rename it to `FileWriteStr`, and pass in the `file` argument explicitly. This makes for a clearer utility function.